src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -91,22 +91,28 @@ export class DatasetService {
   public async fetchOptions(
     services: IDataPluginServices,
     path: DataStructure[],
-    dataType: string
+    dataType: string,
+    options?: DataStructureFetchOptions
```

Comment:
1 more thing. i think @amsiglan  implemented a `getSearchOptions` perhaps we should introduce a `getFetchOptions`

Reply:
i think `fetchOptions` should be defined by the caller of `fetch` to specify whether it wants to fetch next page or with some search keyword, not by the `DatasetTypeConfig`? whereas [`getSearchOptions`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1750cadee4d94c62532ddff377c74180c2b19ceb/src/plugins/data/public/query/query_string/dataset_service/types.ts#L57-L61) should be determined by the type config

i feel the `fetch` vs `search` is getting confusing, `fetch` is actually populating the properties like `children` of the given datastructure. i'll add/update jsdocs